File: Crio/crio_programming_interview_problems-master/BeesToHives/BeesToHives.py

def beesToHives(bees, hives):
# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    n = len(bees)
    bees.sort()
    hives.sort()
    diff = [abs(bees[i]- hives[i]) for i in range(n)]
    ans = max(diff)
    return ans
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS    

def main():
    n = input()
    bees = list(map(int,input().split()))
    hives = list(map(int,input().split()))
    assert len(bees) == len(hives)
    ans = beesToHives(bees, hives)
    print(ans)

if __name__ == "__main__":
    main()

File: Crio/crio_programming_interview_problems-master/BeesToHives/BeesToHives.cpp
#include<bits/stdc++.h>
using namespace std;
int beesToHives(vector<int>bees, vector<int>hives){
    // CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
    int ans = 0;
    sort(bees.begin(),bees.end());
    sort(hives.begin(),hives.end());
    int n = bees.size();
    for (int i = 0; i < n; i++)
    {
        ans = max(ans, abs(bees[i]-hives[i]));
    }
    return ans;
    // CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
}
int main(){
    int n;
    cin>>n;
    vector<int>bees(n),hives(n);
    for (int i = 0; i < n; i++)
    {
        cin>>bees[i];
    }
    for (int i = 0; i < n; i++)
    {
        cin>>hives[i];
    }
    int ans = beesToHives(bees,hives);
    cout<<ans;
}